SF iOS/SF iOS/Feed/EventsFeedViewController.m

```diff
@@ -66,6 +69,84 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
+                                                                   message:nil
+                                                            preferredStyle:UIAlertControllerStyleActionSheet];
+
+    void (^notificationsPreferenceChangeTapped)(UIAlertAction * _Nonnull) = ^void(UIAlertAction * _Nonnull action) {
+
+        // if notifications are turned on, set them to off
+        if (isNotificationSet) {
+            [[NSUserDefaults standardUserDefaults] setNotificationSetting:FALSE
+                                                                 forGroup:nil];
+            [self updateNotificationButton];
+        }
+        // if notifications are turned off, request permissions
+        // only if they are granted do we update the setting
+        else {
+            UNUserNotificationCenter *notificationCenter = [UNUserNotificationCenter currentNotificationCenter];
+            UNAuthorizationOptions options = UNAuthorizationOptionAlert|UNAuthorizationOptionBadge|UNAuthorizationOptionSound;
+            __weak typeof(self) welf = self;
+
+            [notificationCenter requestAuthorizationWithOptions:options
+                                              completionHandler:^(BOOL granted, NSError *error){
+                                                  // Not a very robust error handling solution here
+                                                  if (error) {
+                                                      NSLog(@"Error %@", error);
+                                                  }
+                                                  // if they've granted perms, update the setting, notify UI
+                                                  if (granted) {
+                                                      [[NSUserDefaults standardUserDefaults] setNotificationSetting:!isNotificationSet
+                                                                                                           forGroup:nil];
+                                                      // This hits UI so we must dispatch on main queue
+                                                      dispatch_async(dispatch_get_main_queue(), ^{
+                                                          [welf updateNotificationButton];
+                                                      });
+                                                  }
+                                              }];
+        };
+    };
+    [alert addAction:[UIAlertAction actionWithTitle:buttonTitle
+                                              style:UIAlertActionStyleDefault
+                                            handler:notificationsPreferenceChangeTapped]];
+
+    [alert addAction:[UIAlertAction actionWithTitle:cancelButtonTitle
+                                              style:UIAlertActionStyleCancel
+                                            handler:nil]];
+    [self presentViewController:alert animated:true completion:nil];
```

Comment:
I’d like the UI layering cleaned up a little, please.

Before the alert is show, hide the button. On completion, show the button again.

Reply:
Not sure if I follow. Which button are we talking about? The `UIAction` button or the notification button that was added in this PR? And is the alert you are speaking of the Action Sheet or the system alert asking for notification permissions?

Reply:
Notification button should hide before the action sheet appears and return when the action sheet dismisses.

Reply:
Got it

Reply:
Ok yeah that is a lot better. I pushed the change just now.